pkg/core/proxy/proxy.go

```diff
@@ -729,3 +754,24 @@ func (p *Proxy) GetConsumedMocks(_ context.Context, id uint64) ([]models.MockSta
 	}
 	return m.(*MockManager).GetConsumedMocks(), nil
 }
+
+// GetErrorChannel returns the error channel for external monitoring
+func (p *Proxy) GetErrorChannel() <-chan error {
+	return p.errChannel
+}
+
+// SendError sends an error to the error channel for external monitoring
+func (p *Proxy) SendError(err error) {
+	select {
+	case p.errChannel <- err:
+		// Error sent successfully
+	default:
+		// Channel is full, log the error instead
```

Comment:
Did you encounter this issue? where you were enable to send the error because the channel was full.

Reply:
No, Didn't encounter this issue as the buffered channel limit is 100, and currently just sending the error when the mock outgoing was failing so it will not reach the limit